napari/layers/utils/plane_manager.py

```diff
@@ -77,3 +74,110 @@ def from_points(cls, a, b, c):
+        return np.stack([self.position, self.normal])
+
+    @classmethod
+    def from_array(cls, array, enabled=True):
+        """
+        construct the plane from a (2, 3) array
+        """
+        return cls(
+            position=array[0],
+            normal=array[1],
+            enabled=enabled,
+        )
+
+    def __hash__(self):
+        return id(self)
+
+
+class PlaneList(SelectableEventedList):
+    """
+    A list of planes providing some utility methods
+    """
+
+    def as_array(self):
+        """
+        return a (N, 2, 3) array representing the planes
+        """
+        arrays = []
+        for plane in self:
+            if plane.enabled:
+                arrays.append(plane.as_array())
+        if not arrays:
+            return np.empty((0, 2, 3))
+        return np.stack(arrays)
+
+    @classmethod
+    def from_array(cls, array):
+        """
+        construct the PlaneList from an array of shape (N, 2, 3)
+        """
+        if array.ndim != 3 or array.shape[1:] != (2, 3):
+            raise ValueError(
+                trans._(
+                    'Planes can only be constructed from arrays of shape (N, 2, 3), not {shape}',
+                    deferred=True,
+                    shape=array.shape,
+                )
+            )
+        planes = [PlaneManager.from_array(sub_arr) for sub_arr in array]
+        return cls(planes)
+
+    @classmethod
+    def from_bounding_box(cls, center, dimensions):
+        """
+        generate 6 planes positioned to form a bounding box, with normals towards the center
+
+        Parameters
+        ----------
+        center : ArrayLike
+            (3,) array, coordinates of the center of the box
+        extents : ArrayLike
+            (3,) array, dimensions of the box
+
+        Returns
+        -------
+        list : PlaneList
+        """
+        planes = []
+        for axis in range(3):
+            for direction in (-1, 1):
+                shift = (dimensions[axis] / 2) * direction
+                position = np.array(center)
+                position[axis] += shift
+
+                normal = np.zeros(3)
+                normal[axis] = -direction
+
+                planes.append(
+                    PlaneManager(
+                        position=position, normal=normal, enabled=True
+                    )
+                )
+        return cls(planes)
+
+
+class ThickPlaneManager(PlaneManager):
+    """Defines a thick plane in 3D and provides utility methods and events to handle it.
+
+    Thick planes are defined by a position a normal vector and a thickness.
+
+    Attributes
+    ----------
+    position : 3-tuple
+        A 3D position on the plane, defined in data coordinates.
+    normal : 3-tuple
+        A 3D unit vector normal to the plane, defined in data coordinates.
+    thickness : float
+        Thickness of the slice.
+    enabled : bool
+        Whether the plane is considered enabled.
+    """
+
+    thickness: float = 10.0
```

Comment:
I wonder if we need a new object here, or if we should just allow `thickness = None` for clipping planes. I worry that the additional object causes a bit more overhead when reasoning about things for not too much gained. Thoughts @jni?

Reply:
I also wonder about the naming "manager" ... i see we're calling more and more things "Manager".  What is this managing?  it looks like a pretty simple set of attributes describing a plane, in which case I'd propose we call it `Plane` ... and I agree with @sofroniewn that it doesn't need a new subclass.  I'd vote for `Plane` having a default thickness of 0, and specifying thickness > 0 when you need a "ThickPlane".

Reply:
Yup agreed - these are just planes, let's rename `Plane` and go with default thickness of `0`. Thanks for input @tlambert03!!

Reply:
I switched to manager when I added the `enabled` attribute because it's an object managing the plane rendering but I agree it's not really necessary, let's keep it simple!

Reply:
Yeah, I think we should reserve "manager" for something managing a set of objects (like the old "settings manager" was managing a set of more complicated objects).  If  it's just "managing" a set of attributes that are used somewhere, then i think it's pretty much a pure model object.

Reply:
Agree on everything!